server/src/test/java/io/crate/execution/engine/fetch/FetchTaskTest.java

```diff
@@ -104,12 +116,12 @@ public void testSearcherIsAcquiredForShard() throws Exception {
                 tableIndices,
                 List.of(createReference("i1", new ColumnIdent("x"), DataTypes.STRING))),
             "dummy",
-            new SharedShardContexts(mock(IndicesService.class, RETURNS_MOCKS), UnaryOperator.identity()),
+            sharedShardContexts,
             metadata,
             relationName -> null,
             List.of(routing));
 
-        context.start();
+        context.start().get(5, TimeUnit.SECONDS);
```

Comment:
I'm not sure I understand how the fix is tested, is it easy to provide some more info please?

Reply:
It wasn't tested. I pushed a fixup to at least test that the refreshReaders call happens